File: LEETCODE-CPP/1768.cpp
//1768. Merge Strings Alternately

class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        
        int l1=word1.length(),l2=word2.length();
        string res="";
        
        int i=0,j=0;
        while(i<l1 || j<l2){
            
            if(i<l1){
                res+=word1[i];
                i++;
            }
            
            if(j<l2){
                res+=word2[j];
                j++;
            }
        }
    
        return res;
    }
};

File: LEETCODE/LEETCODE-PYTHON/1768.py
# 1768. Merge Strings Alternately
# https://leetcode.com/problems/merge-strings-alternately

class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        res=[]

        i=0
        while i<len(word1) or i<len(word2):
            if i<len(word1):
                res.append(word1[i])
            if i<len(word2):
                res.append(word2[i])

            i+=1

        return ''.join(res)